lib/matplotlib/cbook/__init__.py

```diff
@@ -1691,19 +1691,12 @@ def safe_first_element(obj):
     This is an type-independent way of obtaining the first element, supporting
     both index access and the iterator protocol.
     """
```

Comment:
OK, but now this is _not_ the first element, its the first non-None element.  I'm not sure it makes sense to do this here...

Reply:
Thank you for the review. Indeed. Then I guess I have to change where it says "first element" to "first non-None element"

Reply:
.. and rename the method or wrap the None check in a new private method that is a level up?  This is a public method and this is theoretically a breaking change for someone using it and the new behaviour no longer matches the method name

Reply:
@jklymak 
Please correct me if I misunderstood this. I guess I should make a new private method(like _safe_first_non_none_element()), keeping the whole old safe_first_element().
Or, I still need to have the original safe_first_element() even if I rename the method.

Reply:
I think

`def safe_first_element(obj, skip_none=False):`

is the best way to go here. We should explicitly opt in to skipping None where needed.

Reply:
I think @timhoffm suggestion makes sense.  We could wrap that in a `_safe_first_non_none` or some such for our internal use if that is easier than adding a kwarg to each call...

Reply:
On more consideration, I agree with @timhoffm .  I thought this was an internal helper not a public.